diff --git a/drivers/net/phy/Kconfig b/drivers/net/phy/Kconfig
index 4e78a4d..44dace2 100644
--- a/drivers/net/phy/Kconfig
+++ b/drivers/net/phy/Kconfig
@@ -263,6 +263,16 @@ config MAXLINEAR_GPHY
 	select POLYNOMIAL if HWMON
 	depends on HWMON || HWMON=n
 	help
+<<<<<<<
+=======
+	  Support for the Maxlinear GPY115, GPY211, GPY212, GPY215,
+	  GPY241, GPY245 PHYs.
+
+source "drivers/net/phy/mediatek/Kconfig"
+
+config MICREL_PHY
+	tristate "Micrel PHYs"
+>>>>>>>
 	  Support for the Maxlinear GPY115, GPY211, GPY212, GPY215,
 	  GPY241, GPY245 PHYs.
 
